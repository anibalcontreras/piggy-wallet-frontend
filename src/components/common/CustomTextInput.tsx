import React from 'react';
import { TextInput, StyleSheet } from 'react-native';
import { Colors, Outlines, Sizing } from '../../styles';

function CustomTextInput({ ...props }): JSX.Element {
  return (
    <TextInput
      style={[styles.input]}
      placeholderTextColor={Colors.palette.primary}
      keyboardAppearance="dark"
      selectionColor={Colors.palette.primary}
      {...props}
    />
  );
}

const styles = StyleSheet.create({
  input: {
    height: Sizing.layout.x60,
    width: Sizing.screen.width - Sizing.layout.x30,
    padding: Sizing.layout.x20,
    marginBottom: Sizing.layout.x20,
    borderColor: Colors.palette.border,
    borderRadius: Outlines.borderRadius.base,
    borderWidth: Outlines.borderWidth.thin,
    color: Colors.palette.primary,
  },
});

export default CustomTextInput;
