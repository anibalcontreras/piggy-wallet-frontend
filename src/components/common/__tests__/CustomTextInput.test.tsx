import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import CustomTextInput from '../CustomTextInput';

describe('CustomTextInput Component', () => {
  it('renders custom text input correctly', () => {
    const tree = renderer.create(<CustomTextInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes the correct props to TextInput', () => {
    const placeholderText = 'Test Placeholder';
    const tree = renderer.create(<CustomTextInput placeholder={placeholderText} />).toJSON();
    expect(tree).toHaveProperty(['props', 'placeholder'], placeholderText);
  });

  it('handles onChangeText event', () => {
    const mockOnChangeText = jest.fn();
    const { getByPlaceholderText } = render(
      <CustomTextInput onChangeText={mockOnChangeText} placeholder="Test Placeholder" />
    );

    fireEvent.changeText(getByPlaceholderText('Test Placeholder'), 'New Text');
    expect(mockOnChangeText).toHaveBeenCalledWith('New Text');
  });
});
