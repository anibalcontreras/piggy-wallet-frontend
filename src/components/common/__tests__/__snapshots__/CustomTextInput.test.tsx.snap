// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextInput Component renders custom text input correctly 1`] = `
<TextInput
  keyboardAppearance="dark"
  placeholderTextColor="#01C38D"
  selectionColor="#01C38D"
  style={
    [
      {
        "borderColor": "#696E79",
        "borderRadius": 10,
        "borderWidth": 1,
        "color": "#01C38D",
        "height": 50,
        "marginBottom": 18,
        "padding": 18,
        "width": 724,
      },
    ]
  }
/>
`;
